db/migrate/20200512044127_change_messages_to_polymorphic.rb

```diff
@@ -0,0 +1,46 @@
+class ChangeMessagesToPolymorphic < ActiveRecord::Migration
+  def up
+    add_column :messages, :messageable_type, :string
+    add_column :messages, :messageable_id, :int
+    Message.all.map do |message|
+      if message.item_id
+        item = message.item
+        message.messageable_type = item.class.name
+        message.messageable_id = item.id
+      elsif message.offer_id
+        offer = message.offer
+        message.messageable_type = offer.class.name
+        message.messageable_id = offer.id
+      elsif message.order_id
+        order = message.order
+        message.messageable_type = order.class.name
+        message.messageable_id = order.id
+      end
+      message.save!(validate: false)
+    end
+
+    remove_column :messages, :offer_id
+    remove_column :messages, :order_id
+    remove_column :messages, :item_id
+  end
+
+  def down
+    add_column :messages, :offer_id, :int
+    add_column :messages, :order_id, :int
+    add_column :messages, :item_id, :int
+
+    Message.all.map do |message|
+      case message.messageable_id
```

Comment:
this should be `message.messageable_type` but would suggest rewriting the migration in an SQL way using a similar method I posted in comments for `up`